Homework_8_RPG_Game

ira-01 opened the pull request.

ira-01 committed:
Homework_8_RPG_Game

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/Character.java:
```diff
@@ -0,0 +1,66 @@
+
+
+
+    public int Attack(Character otherPerson) {
```
Non-static methods of the class - names should start with a lowercase letter according to convention

`public int attack`

Only constructor methods start with capital letter (as they have to be named identically to how their class is named)


https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/Character.java:
```diff
@@ -0,0 +1,66 @@
+                 isAlive = false;
+            }
+
+       return isAlive;
```
works well, although the whole body can be simplified to 

`return this.health > 0;`

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/Character.java:
```diff
@@ -0,0 +1,66 @@
+            if (damage < 0) {
+                damage = 0;
+            }
+
```
here one thing is missing:
You have `otherPerson` (enemy) object passed as a method parameter, so you can directly call

`otherPerson.takeDamage(damage)`

so you won't be forced to call it explicitly in the client code (Main class)

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/Main.java:
```diff
@@ -0,0 +1,36 @@
+            selected = select.nextBoolean();
+
+            if (selected = true) {
+                firstPerson.takeDamage(secondPerson.Attack(firstPerson));
```
if  `.attack()` method included other person taking damage automatically - the whole construction would be much more simple

`firstPerson.attack(secondPerson)`

The inflicted damage would be automatically calculated and `secondPerson.takeDamage()` would be implicitly called inside the `.attack()` method body

rodikno reviewed (changes requested):
Requesting some changes: variable and method naming according to convention + OOP-style followed while defining the `.attack(otherPerson)` method

ira-01 committed:
Minor update

Made changes in the task.

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/character.java:
```diff
@@ -0,0 +1,57 @@
+public class character {
```
По конвенції назви класів пишемо з великої букви: `Character`

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/character.java:
```diff
@@ -0,0 +1,57 @@
+            }
+   }
+
+    protected boolean isAlive (){
```
Ctrl + Alt + L :)

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/character.java:
```diff
@@ -0,0 +1,57 @@
+           System.out.println(this.name + " strenght = " + this.strength); //strenght
+           System.out.println(this.name + " defense = " + this.defense); //defense
+       }
+       else{
```
Ctrl + Alt + L

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/character.java:
```diff
@@ -0,0 +1,57 @@
+        int damage;
+        damage = this.strength - otherPerson.defense;
+
+            if (damage < 0) {
```
Ctrl + Alt + L

rodikno commented on Homework_8_MyAmazingRPG/MyAmazingRPG/src/Main.java:
```diff
@@ -0,0 +1,36 @@
+        firstPerson.displayStatus();
+        secondPerson.displayStatus();
+        System.out.println("Person count = " + character.characterCount);
+        System.out.println("GAME OVER");
```
Who won? :)

ira-01 committed:
Minimal changes made

Minimal changes have been made, including the addition of a winner announcement after the battle concludes.
